frame/staking/src/lib.rs

```diff
@@ -1163,7 +1191,7 @@ decl_module! {
 			T::SlashCancelOrigin::try_origin(origin)
 				.map(|_| ())
 				.or_else(ensure_root)
-				.map_err(|_| "bad origin")?;
+				.map_err(|_| Error::BadOrigin)?;
```

Comment:
```suggestion
				.map_err(|_| Error::InvalidOrigin)?;
```

Reply:
I'd prefer for these general bad origin errors we just had one global error that everything can reference and that `try_origin` can return.

Reply:
https://github.com/paritytech/substrate/issues/4409 - these grumbles can be fixed with that.